server/test/controllers/api/ProgramApplicationsApiControllerTest.java

```diff
@@ -0,0 +1,142 @@
+package controllers.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertThrows;
+import static play.api.test.Helpers.testServerPort;
+import static play.test.Helpers.contentAsString;
+import static play.test.Helpers.fakeRequest;
+import static play.test.Helpers.route;
+
+import auth.ApiKeyGrants;
+import auth.UnauthorizedApiRequestException;
+import com.jayway.jsonpath.DocumentContext;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+import models.ApiKey;
+import models.Program;
+import org.apache.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Test;
+import play.mvc.Http;
+import play.mvc.Result;
+import services.applicant.JsonPathProvider;
+import services.export.AbstractExporterTest;
+
+public class ProgramApplicationsApiControllerTest extends AbstractExporterTest {
+
+  private ApiKey apiKey;
+  private static final String keyId = "key-id";
+  private static final String keySecret = "key-secret";
+  private static final String rawCredentials = keyId + ":" + keySecret;
+  private static final String serializedApiKey =
+      Base64.getEncoder().encodeToString(rawCredentials.getBytes(StandardCharsets.UTF_8));
+
+  @Before
+  public void setUp() {
+    apiKey = resourceCreator.createActiveApiKey("test-key", keyId, keySecret);
+  }
+
+  @Test
+  public void list_success_dateRange() {
+    createFakeProgramWithEnumerator();
+    apiKey
+        .getGrants()
+        .grantProgramPermission(fakeProgramWithEnumerator.getSlug(), ApiKeyGrants.Permission.READ);
+    apiKey.save();
+
+    String requestUrl =
+        controllers.api.routes.ProgramApplicationsApiController.list(
+                fakeProgramWithEnumerator.getSlug(),
+                /* fromDate= */ Optional.of("2022-01-15"),
+                /* toDate= */ Optional.of("2022-02-15"),
+                /* nextPageToken= */ Optional.empty(),
+                /* pageSize= */ Optional.empty())
+            .url();
+
+    Result result = doRequest(requestUrl);
+    assertThat(result.status()).isEqualTo(HttpStatus.SC_OK);
+
+    DocumentContext resultJson = JsonPathProvider.getJsonPath().parse(contentAsString(result));
+    assertThat(resultJson.read("payload[0].application_id", Long.class))
+        .isEqualTo(applicationTwo.id);
+    assertThat(resultJson.read("payload.length()", Integer.class)).isEqualTo(1);
+  }
+
+  @Test
+  public void list_success_multiplePages() {
```

Comment:
This is really two tests, and it'd be clearer to break them up.

1. multiple results in a page.
2. multiple pages (this doesn't need to verify the first page because of the above and existing tests.

Reply:
done